src/message_conversions.rs

```diff
@@ -0,0 +1,530 @@
+                    .set_return_code(vsomeip::return_code_e::E_NOT_OK);
+                get_pinned_message_base(&vsomeip_msg).set_message_type(message_type_e::MT_ERROR);
+            }
+
+            trace!(
+                "{} - Response: Finished building vsomeip message: service_id: {} instance_id: {}",
+                UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_UMSG_TO_VSOMEIP_MSG,
+                service_id,
+                instance_id
+            );
+
+            Ok(vsomeip_msg)
+        }
+        _ => Err(UStatus::fail_with_code(
+            UCode::INTERNAL,
+            "Trying to convert an unspecified or notification message type.",
+        )),
+    }
+}
+
+pub async fn convert_vsomeip_msg_to_umsg(
+    vsomeip_message: &mut UniquePtr<MessageWrapper>,
+    _application_wrapper: &UniquePtr<ApplicationWrapper>,
+    _runtime_wrapper: &UniquePtr<RuntimeWrapper>,
+) -> Result<UMessage, UStatus> {
+    trace!("top of convert_vsomeip_msg_to_umsg");
+    let msg_type = get_pinned_message_base(vsomeip_message).get_message_type();
+
+    let request_id = get_pinned_message_base(vsomeip_message).get_request();
+    let service_id = get_pinned_message_base(vsomeip_message).get_service();
+    let client_id = get_pinned_message_base(vsomeip_message).get_client();
+    let session_id = get_pinned_message_base(vsomeip_message).get_session();
+    let method_id = get_pinned_message_base(vsomeip_message).get_method();
+    let instance_id = get_pinned_message_base(vsomeip_message).get_instance();
+    let interface_version = get_pinned_message_base(vsomeip_message).get_interface_version();
+    let payload = get_message_payload(vsomeip_message);
+    let payload_bytes = get_data_safe(&payload);
+
+    trace!("{} - : request_id: {} client_id: {} session_id: {} service_id: {} instance_id: {} method_id: {} interface_version: {}",
+        UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_VSOMEIP_MSG_TO_UMSG,
+        request_id, client_id, session_id, service_id, instance_id, method_id, interface_version
+    );
+
+    let authority_name = { AUTHORITY_NAME.lock().await.clone() };
+
+    trace!("unloaded all relevant info from vsomeip message");
+
+    match msg_type {
+        message_type_e::MT_REQUEST => {
+            trace!("MT_REQUEST type");
+            let sink = UUri {
+                authority_name,
+                ue_id: service_id as u32,
+                ue_version_major: interface_version as u32,
+                resource_id: method_id as u32,
+                ..Default::default()
+            };
+
+            let source = UUri {
+                authority_name: ME_AUTHORITY.to_string(), // TODO: Should we set this to anything specific?
+                ue_id: client_id as u32,
+                ue_version_major: 1, // TODO: I don't see a way to get this
+                resource_id: 0,      // set to 0 as this is the resource_id of "me"
+                ..Default::default()
+            };
+
+            // TODO: Not sure where to get this
```

Comment:
I'm not sure where to get a TTL value for a uProtocol Request message when translating from an MT_REQUEST vsomeip message. It would appear there's not a TTL value in the SOME/IP header.

Any ideas @stevenhartley / @int0x27?